### DISCUSSION TITLE
Current round effort

### DISCUSSION THREAD

**Discussion by riksarchen** [coinfoundry/miningcore#963] [Q&A]
Hello 
Is there anyway to get current round effort and estimated payout for miner if block is found now from current api?

**Comment by riksarchen** [comment]
I had a look at the code and it seems to be doable to add without much modification. 

Added data would be nice:
Timestamp when last share was submitted per worker.

Total number of shares submitted from all workers since last block was found.

Total number of shares submitted per worker since last block was found.

With this data current round effort could be calculated and to display when worker last submitted share with notification if time is above a predetermined value for worker offline notifications

**Comment by jvxis** [comment]
Did anyone figure out how to implement Round Effort?
Thanks
Jaime

**Reply by honorkeles** [reply]
I am trying to figure out a way to add this to api as well.

I think we need to get api to generate current round total shares to get the Round effort. 

we can use below once we have current round total shares from the api

          var effort = (
            (value.poolStats.sharesDiff /
              value.networkStats.networkDifficulty) *
            100
          ).toFixed(2);

thanks 
Onur

**Reply by jvxis** [reply]
Hi, I think you need to sum all miners pendings shares and then divide by Network difficulty!

**Reply by honorkeles** [reply]
> Hi, I think you need to sum all miners pendings shares and then divide by Network difficulty!

miners pendings share is not provided from the api as well. I think I getting miners pending shares from the api would be best way which will fix couple issues

thanks

**Reply by jvxis** [reply]
Yes, it is provided. You need to call each miner dashboard (miner/performance) and sum.

**Reply by jvxis** [reply]
https://github.com/oliverw/miningcore/wiki/API#api-pools-miners-performance

**Reply by honorkeles** [reply]
> Yes, it is provided. You need to call each miner dashboard (miner/performance) and sum.

I was trying to call api for all possible miners with over 0 pendingshare but couldn't figure it out. 

Do you have any quick solution to this?

thanks

**Reply by jvxis** [reply]
Unfortunately no so far.

**Comment by honorkeles** [comment]
Thank you

**Comment by jvxis** [comment]
Hi @oliverw, I think this feature is important to be prioritized. It brings transparency for miners regarding the current effort.

**Comment by jon4hz** [comment]
This feature can be backported from https://github.com/GetBlok-io/miningcore/blob/master/src/Miningcore/Persistence/Postgres/Repositories/ShareRepository.cs#L203

starttime would be the find time from the last block and endtime would be now.

**Reply by oliverw** [reply]
> This feature can be backported from https://github.com/GetBlok-io/miningcore/blob/master/src/Miningcore/Persistence/Postgres/Repositories/ShareRepository.cs#L203

PR? 😁
